receiver/simpleprometheusreceiver/config.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package simpleprometheusreceiver
+
+import (
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
+)
+
+// Config defines configuration for simple prometheus receiver.
+type Config struct {
+	configmodels.ReceiverSettings `mapstructure:",squash"`
+	// CollectionInterval is the interval at which metrics should be collected
+	CollectionInterval time.Duration `mapstructure:"collection_interval"`
+	// MetricsPath the path to the metrics endpoint.
+	MetricsPath string `mapstructure:"metrics_path"`
+	// Whether or not to use pod service account to authenticate.
+	UseServiceAccount bool `mapstructure:"use_service_account"`
+	// TLS configs in case connection is over HTTPS
+	TLSCredentials *tlsCredentials `mapstructure:"tls_credentials"`
```

Comment:
I think I might just call the config value `tls` as they're not necessarily credentials. For example you might set ca_file but not use cert_file and key_file for auth.

Reply:
What does this look like in config if you want to enable TLS but with just the defaults? If you do:

```yaml
receivers:
  prometheus_simple
    tls:
```

I think tls will be nil so not configured. But if you did:

```yaml
receivers:
  prometheus_simple
    tls: {}
```

TLS would be non-nil with the default zero values. Seems like it might be easy to mess up. 🤔

Reply:
Since paths to the cert, key and the CA cert are all empty an empty TLS config would fail authentication right? Also, would it be a valid use case to have these be the empty default values?

Reply:
If it's empty it should use the system CA certs.

Reply:
Looks like they avoid this problem by having TLS explicitly enabled or disabled with `scheme`.

Reply:
Oh ok. I just looked at how prometheus handles these values and from first look, it does not seem like it picks up the system CA certs. Will dig a bit deeper and get back.

Reply:
I think it'd end up using the default which is system default CA:

https://golang.org/src/crypto/tls/common.go?s=16110:25304#L526

```go
	// RootCAs defines the set of root certificate authorities
	// that clients use when verifying server certificates.
	// If RootCAs is nil, TLS uses the host's root CA set.
	RootCAs *x509.CertPool
```